content/docs/installation/best-practice.md

```diff
@@ -22,6 +22,174 @@ are designed for backwards compatibility rather than for best practice or maximu
+> But the controller manager works asynchronously anyway, so any applications which depend on the cert-manager custom resources
+> will be designed to tolerate this situation.
+> That being said, the best practice is to run 2 or more replicas of each controller if the cluster has sufficient resources.
+
+### Topology Spread Constraints
+
+Ensure that a disruption of a node or data center does not degrade the operation of cert-manager,
+by configuring [Topology Spread Constraints](https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/)
+for each of the components.
+For example, the following Helm chart values add topology spread constraints for all three long-running components,
+to request (but not require) Kubernetes to avoid scheduling Pods of the same Deployment to the same zone or node.
+
+```yaml
+topologySpreadConstraints:
+- maxSkew: 1
+  topologyKey: topology.kubernetes.io/zone
+  whenUnsatisfiable: ScheduleAnyway
+  labelSelector:
+    matchLabels:
+      app.kubernetes.io/instance: cert-manager
+      app.kubernetes.io/component: controller
+- maxSkew: 1
+  topologyKey: kubernetes.io/hostname
+  whenUnsatisfiable: ScheduleAnyway
+  labelSelector:
+    matchLabels:
+      app.kubernetes.io/instance: cert-manager
+      app.kubernetes.io/component: controller
+
+webhook:
+  replicaCount: 3
+  topologySpreadConstraints:
+  - maxSkew: 1
+    topologyKey: topology.kubernetes.io/zone
+    whenUnsatisfiable: ScheduleAnyway
+    labelSelector:
+      matchLabels:
+        app.kubernetes.io/instance: cert-manager
+        app.kubernetes.io/component: webhook
+  - maxSkew: 1
+    topologyKey: kubernetes.io/hostname
+    whenUnsatisfiable: ScheduleAnyway
+    labelSelector:
+      matchLabels:
+        app.kubernetes.io/instance: cert-manager
+        app.kubernetes.io/component: webhook
+
+cainjector:
+  replicaCount: 2
+  topologySpreadConstraints:
+  - maxSkew: 1
+    topologyKey: topology.kubernetes.io/zone
+    whenUnsatisfiable: ScheduleAnyway
+    labelSelector:
+      matchLabels:
+        app.kubernetes.io/instance: cert-manager
+        app.kubernetes.io/component: cainjector
+  - maxSkew: 1
+    topologyKey: kubernetes.io/hostname
+    whenUnsatisfiable: ScheduleAnyway
+    labelSelector:
+      matchLabels:
+        app.kubernetes.io/instance: cert-manager
+        app.kubernetes.io/component: cainjector
+```
+
+### PodDisruptionBudget
+
+For high availability you should also deploy a `PodDisruptionBudget` resource,
+with `minAvailable=1` *or* with `maxUnavailable=1`.
+This ensures that a *voluntary* disruption, such as the draining of a Node, can not proceed
+until at least one other replica has been successfully scheduled and started on another Node.
+The Helm chart has parameters to enable and configure a PodDisruptionBudget
+for each of the long-running cert-manager components.
+
+
```

Comment:
Perhaps include the relevant subset of YAML to configure PDBs? As you have done with the topology settings?

```yaml
podDisruptionBudget:
  enabled: true
  minAvailable: 1
webhook:
  podDisruptionBudget:
    enabled: true
    minAvailable: 1
cainjector:
  podDisruptionBudget:
    enabled: true
    minAvailable: 1
```

Reply:
Done.